### PR TITLE
added some routes and middleware

### PR DESCRIPTION
**Pull request by Abigail-cloud**
added Item functionality

### CODE REVIEW

File: src/Controllers/ItemController/item.ts [hunk 1]
```diff
@@ -22,3 +26,24 @@ export const createItem = (req: Request, res: Response) => {
             res.status(400).json(error);
         });
 };
+
+
+
+export const getTotalItems = async (req: Request, res: Response) => {
+   const items = await getAllItems(req, res);
```
Comment (forestbaba) [inline src/Controllers/ItemController/item.ts]: This needs to be paginated

File: src/Repository/ItemRepository/item.ts [hunk 2]
```diff
@@ -13,4 +13,24 @@ export class ItemRepository {
...
+  }
+ 
+  public async editItem(itemId: any): Promise<any> {
+    const item = await Item.findById(itemId)
+    return item
+  }
+
+  public async removeItem(Id: any): Promise<any> {
+    const item= await Item.findById(Id)
+    if (!item) return null;
```
Comment (forestbaba) [inline src/Repository/ItemRepository/item.ts]: You don't need to check for nullity here..Just return the item. Check for nullity in service

File: src/services/ItemService/item.ts [hunk 3]
```diff
@@ -14,3 +16,71 @@ export const createItemService = async (newItem: ItemType) => {
...
+    name,
+    price,
+    description,
+    image,
+    location,
+    category,
+    status,
+    shippingOptions,
+  } = req.body
+console.log(req.body);
```
Comment (forestbaba) [inline src/services/ItemService/item.ts]: Remove this console log

File: src/services/ItemService/item.ts [hunk 4]
```diff
@@ -14,3 +16,71 @@ export const createItemService = async (newItem: ItemType) => {
...
+
+  const item = await new ItemRepository().editItem (req.params.id)
+
+  if (item) {
+    item.name = name
```
Comment (forestbaba) [inline src/services/ItemService/item.ts:50]: What if the name and other details the user is parsing during update is empty, did you cater for that in your middleware??

File: src/services/ItemService/item.ts [hunk 5]
```diff
@@ -14,3 +16,71 @@ export const createItemService = async (newItem: ItemType) => {
...
+    const updatedProduct = await item.save()
+    res.json(updatedProduct)
+  } else {
+    res.status(404)
+    throw new NotFoundError(' Item not available')
+  }
+}
+
+export const deleteItem = async (req:Request, res:Response) => {
+  const item = await new ItemRepository().removeItem(req.params.id)
```
Comment (forestbaba) [inline src/services/ItemService/item.ts]: Make this soft delete instead

**Review by forestbaba — COMMENTED** [review]
Please remove your beautiful face file and all files you added

File: src/services/ItemService/item.ts (hunk 4 above)
Comment (Abigail-cloud) [reply src/services/ItemService/item.ts:50]: Added...

File: src/services/ItemService/item.ts (hunk 5 above)
Comment (Abigail-cloud) [reply src/services/ItemService/item.ts]: done

File: src/services/ItemService/item.ts (hunk 3 above)
Comment (Abigail-cloud) [reply src/services/ItemService/item.ts]: Done

File: src/Repository/ItemRepository/item.ts (hunk 2 above)
Comment (Abigail-cloud) [reply src/Repository/ItemRepository/item.ts]: Done

File: src/Controllers/ItemController/item.ts (hunk 1 above)
Comment (Abigail-cloud) [reply src/Controllers/ItemController/item.ts]: The pagination was added as a middleware in the routes folder..

**Review by forestbaba — APPROVED** [review]
LGTM

File: src/Controllers/ItemController/item.ts (hunk 1 above)
Comment (forestbaba) [reply src/Controllers/ItemController/item.ts]: Ok

**Pull request merged**